src/nvim/fileio.c

```diff
@@ -5302,32 +5303,43 @@ static void vim_maketempdir(void)
   static const char *temp_dirs[] = TEMP_DIR_NAMES;
```

Comment:
slightly out of scope, but wouldn't using [uv_os_tmpdir](http://docs.libuv.org/en/v1.x/misc.html?highlight=os_tmpdir#c.uv_os_tmpdir) simplify things here?

```suggestion
  char tmpdir[TEMP_FILE_PATH_MAXLEN];
  int r = uv_os_tmpdir(tmpdir, TEMP_FILE_PATH_MAXLEN);

  if (r > 0) {
    // success, tmpdir exists
  }
```

Reply:
uv_os_tmpdir looks nice! But then TMPDIR wouldn't be tried on Windows first, which removes a feature and removes consistency across platforms. That's only the first issue I notice.